driver-core/src/main/com/mongodb/client/model/search/CompoundSearchOperatorBase.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.client.model.search;
+
+import com.mongodb.annotations.Evolving;
+
+/**
+ * @see SearchOperator#compound()
+ * @since 4.7
+ */
+@Evolving
+public interface CompoundSearchOperatorBase {
```

Comment:
I don't see a need for it in the current design (maybe it is a vestige of a previous one).  On my local copy I was able to pull up all the methods from this interface into CompoundSearchOperator and (after a few small changes) remove this interface entirely.